sdks/python/apache_beam/io/gcp/datastore/v1new/helper.py

```diff
@@ -100,6 +100,10 @@ def write_mutations(batch, throttler, rpc_stats_callback, throttle_delay=1):
 
   try:
     start_time = time.time()
+    if batch._status == batch._FINISHED:
```

Comment:
This change is unsafe, since it's dealing with the internal state of the client and the behavior may change unexpectedly in the future.
I believe that a better choice is to create a new batch in this method every time.

Reply:
Agree. I think it's a very good idea to propagate this changes to datastoreio.py as well and avoid building `_batch` there at all, and just accumulate `_mutations` or `_batch_mutations` which we would pass to `write_mutations` and build a new batch every time. This will probably involve test changes as well.

Happy to work on it but would like to confirm this idea.

Reply:
Sounds good